### PR TITLE
ECCW-574: Dockerfiles, .gitlab-ci.yml, docker-compose.yml, oh my...

### PR DESCRIPTION
**Pull request by siliconmeadow**
- [x] Add build and push job (according to gitlab)
- [x] Remove reference to any jobs or dependencies on `mig`
- [x] Remove reference to any jobs or dependencies on `redirect`

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+  needs: ["deploy-to-dev"]
+  environment:
+    name: preproduction
+    url: https://essex-gov-pp.nomensa.xyz/
+  before_script:
+    - az extension add --name containerapp
+    - az login --service-principal -u${ARM_CLIENT_ID} -p${ARM_CLIENT_SECRET} -t${ARM_TENANT_ID}
+  script:
+    - echo "Importing image from Development  into Pre-Production"
+    - az acr import --force --name acreccukspre --source acreccuksdev.azurecr.io/microsites-nginx-drupal:${CI_COMMIT_REF_SLUG} --image microsites-nginx-drupal:${CI_COMMIT_REF_SLUG} --subscription "Essex County Council (Common)"
```
Comment (siliconmeadow) [inline .gitlab-ci.yml:146]: @MatthewWilkes should the subscription be `--subscription "Essex County Council (Microsites)"` here? It was copied and pasted from .gov's gitlab-ci file.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    - az acr import --force --name acreccukspre --source acreccuksdev.azurecr.io/microsites-drupal-fpm:${CI_COMMIT_REF_SLUG} --image microsites-drupal-fpm:${CI_COMMIT_REF_SLUG} --subscription "Essex County Council (Common)"
```
Comment (siliconmeadow) [inline .gitlab-ci.yml:147]: @MatthewWilkes should the subscription be `--subscription "Essex County Council (Microsites)"` here? It was copied and pasted from .gov's gitlab-ci file.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    - az acr import --force --name acreccukspre --source acreccuksdev.azurecr.io/microsites-nginx-redirect:${CI_COMMIT_REF_SLUG} --image microsites-nginx-redirect:${CI_COMMIT_REF_SLUG} --subscription "Essex County Council (Common)"
```
Comment (siliconmeadow) [inline .gitlab-ci.yml]: @MatthewWilkes should the subscription be `--subscription "Essex County Council (Microsites)"` here? It was copied and pasted from .gov's gitlab-ci file.

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+  needs: ["deploy-to-preprod"]
+  environment:
+    name: production
+    url: https://essex.gov.uk/
+  before_script:
+    - az extension add --name containerapp
+    - az login --service-principal -u${ARM_CLIENT_ID} -p${ARM_CLIENT_SECRET} -t${ARM_TENANT_ID}
+  script:
+    - echo "Importing image from Pre-Production into Production"
+    - az acr import --force --name acreccuksprod --source acreccukspre.azurecr.io/microsites-nginx-drupal:${CI_COMMIT_REF_SLUG} --image microsites-nginx-drupal:${CI_COMMIT_REF_SLUG} --subscription "Essex County Council (Common)"
```
Comment (siliconmeadow) [inline .gitlab-ci.yml:218]: @MatthewWilkes should the subscription be `--subscription "Essex County Council (Microsites)"` here? I was copied and pasted from .gov's gitlab-ci file.

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+    - az acr import --force --name acreccuksprod --source acreccukspre.azurecr.io/microsites-drupal-fpm:${CI_COMMIT_REF_SLUG} --image microsites-drupal-fpm:${CI_COMMIT_REF_SLUG} --subscription "Essex County Council (Common)"
```
Comment (siliconmeadow) [inline .gitlab-ci.yml:219]: @MatthewWilkes should the subscription be `--subscription "Essex County Council (Microsites)"` here? It was copied and pasted from .gov's gitlab-ci file.

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+    - az acr import --force --name acreccuksprod --source acreccukspre.azurecr.io/microsites-nginx-redirect:${CI_COMMIT_REF_SLUG} --image microsites-nginx-redirect:${CI_COMMIT_REF_SLUG} --subscription "Essex County Council (Common)"
```
Comment (siliconmeadow) [inline .gitlab-ci.yml]: @MatthewWilkes should the subscription be `--subscription "Essex County Council (Microsites)"` here? It was copied and pasted from .gov's gitlab-ci file.

File: containers.yaml
Comment (siliconmeadow) [inline containers.yaml:1]: @MatthewWilkes - Is this file necessary? I was looking for references about it and some of the vars _did_ make sense in that I'd found them in TF, but others, not so much. (Copied from .gov) btw.

File: containers.yaml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  ipAddress:
+    type: Public
+    ports:
+    - protocol: tcp
+      port: 443
+    dnsNameLabel: ecc-microsites-drupal-dev
+  imageRegistryCredentials:
+  - server: eccgovreg.azurecr.io
+    username: eccmicrositesreg
+    password: <PASSWORD>
```
Comment (siliconmeadow) [inline containers.yaml]: @MatthewWilkes - Over in the [gov repo](https://github.com/essexcountycouncil/essex-gov-uk-drupal/blob/develop/containers.yaml#L46) the closing `>` is missing, which also made me wonder if this file is in use any longer in the gov repo.

File: .gitlab-ci.yml [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+      if [[ "$CI_COMMIT_BRANCH" == "$CI_DEFAULT_BRANCH" ]]; then
+        tag=""
+        echo "Running on default branch '$CI_DEFAULT_BRANCH': tag = 'latest'"
+      else
+        tag=":$CI_COMMIT_REF_SLUG"
+        echo "Running on branch '$CI_COMMIT_BRANCH': tag = $tag"
+      fi
+    - docker pull ${DOCKER_REGISTRY}/microsites-nginx-drupal || true
+    - docker pull ${DOCKER_REGISTRY}/microsites-drupal-fpm || true
+    - docker pull ${DOCKER_REGISTRY}/microsites-nginx-redirect || true
```
Comment (MatthewWilkes) [inline .gitlab-ci.yml]: We don't have a redirect app for microsites

File: .gitlab-ci.yml [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+    - docker build -t ${DOCKER_REGISTRY}/microsites-drupal-fpm${tag} -t microsites-drupal-fpm -f Dockerfile-drupal .
+    - docker build -t ${DOCKER_REGISTRY}/microsites-nginx-drupal${tag} -t microsites-nginx-drupal -f Dockerfile-nginx .
+    - docker build -t ${DOCKER_REGISTRY}/microsites-nginx-redirect${tag} -t microsites-nginx-redirect -f Dockerfile-nginx-redirect .
+    - docker push ${DOCKER_REGISTRY}/microsites-nginx-drupal${tag}
+    - docker push ${DOCKER_REGISTRY}/microsites-drupal-fpm${tag}
+    - docker push ${DOCKER_REGISTRY}/microsites-nginx-redirect${tag}
```
Comment (MatthewWilkes) [inline .gitlab-ci.yml]: Or here

File: .gitlab-ci.yml (hunk 1 above)
Comment (MatthewWilkes) [reply .gitlab-ci.yml:146]: No, this is referencing the container registries, which are common across all three sites.

File: .gitlab-ci.yml (hunk 2 above)
Comment (MatthewWilkes) [reply .gitlab-ci.yml:147]: Here too

File: .gitlab-ci.yml (hunk 3 above)
Comment (MatthewWilkes) [reply .gitlab-ci.yml]: This one should be removed, as there's no redirect app in microsites

File: .gitlab-ci.yml [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+              resources:
+                cpu: 0.25
+                memory: 0.5Gi
+              volumeMounts:
+              - mountPath: /drupal/web/sites/default/files
+                volumeName: filesharevol
+              env:
+              - name: X_ROBOTS_TAG
+                value: noindex
+              - name: LIMITED_BETA_MODE
```
Comment (MatthewWilkes) [inline .gitlab-ci.yml]: Limited beta mode isn't relevant for microsites

File: .gitlab-ci.yml (hunk 4 above)
Comment (MatthewWilkes) [reply .gitlab-ci.yml:218]: As above, this is right.

File: .gitlab-ci.yml (hunk 5 above)
Comment (MatthewWilkes) [reply .gitlab-ci.yml:219]: As above, this is right.

File: .gitlab-ci.yml (hunk 6 above)
Comment (MatthewWilkes) [reply .gitlab-ci.yml]: As above, this should be removed.

File: .gitlab-ci.yml [hunk 11]
```diff
@@ -0,0 +1,268 @@
...
```
Comment (MatthewWilkes) [inline .gitlab-ci.yml]: Limited beta should go in prod also

File: .gitlab-ci.yml [hunk 12]
```diff
@@ -0,0 +1,268 @@
...
+                value: mariadb-root
+              - name: MYSQL_DATABASE
+                value: drupal_microsites
+              - name: MYSQL_PASSWORD
+                secretRef: mysql-password
+              - name: OPENID_CONNECT_PARAMS
+                secretRef: openid-connect-params
+      EOF
+    - az containerapp revision copy --name microsites --resource-group rg-ecc-microsites-uks-prod --yaml revision.yml --subscription "Essex County Council (Microsites)"
+    - az containerapp revision copy --name microsites-nginx-redirect --resource-group rg-ecc-microsites-redirect-uks-prod --image acreccuksprod.azurecr.io/microsites-nginx-redirect:${CI_COMMIT_REF_SLUG} --subscription "Essex County Council (Microsites)"
```
Comment (MatthewWilkes) [inline .gitlab-ci.yml]: No need for the 2nd revision copy for the redirect

File: containers.yaml
Comment (MatthewWilkes) [reply containers.yaml:1]: This is the dockerfile for the nginx container, the ENV lines here set up the defaults if none are provided, and copies the nginx config into the container.

File: containers.yaml (hunk 7 above)
Comment (MatthewWilkes) [reply containers.yaml]: This isn't in use anymore, you can remove it.

**Review by MatthewWilkes — CHANGES_REQUESTED** [review]
Some minor tweaks

**Review by MatthewWilkes — APPROVED** [review]
LGTM
